Add silent option

andersfugmann opened the pull request:
This PR adds option `--silent` which will suppress output for all successful tests.
It contains two commits - one which removes an unused global mutable variable (quiet), and one to add option `--silent`
This addresses #146. 

I did not want this PR to be too intrusive, so there are some `if t.silent then ...` scattered around the code.  Also I would have preferred to not have the first two lines printed (see examples below), but these are printed before any command line arguments are parsed, and I did not want to change the current flow.

Example when all tests are successful:
```
$ dune exec drivers/yaml/test/unittest.exe -- --silent
Testing yaml.
This run has ID `E09E988F-081E-4894-A7CB-E2D3FCFB60A5`.
...............................................................
```

Example with one failing test:
```
$ dune exec drivers/yaml/test/unittest.exe -- --silent
Testing yaml.      
This run has ID `835BF394-F1E5-476D-8A51-14C558AC4D4D`.
................
[ERROR]             Maybe fail                      0   Fail test case.
.............................................
-- Maybe fail.000 [Fail test case.] Failed --
in /home/afu/git/ppx_protocol_conv/_build/_tests/835BF394-F1E5-476D-8A51-14C558AC4D4D/Maybe fail.000.output:
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
ASSERT Blah
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
[failure] Error Blah: expecting
0, got
1.
Raised at file "stdlib.ml", line 33, characters 17-33
Called from file "src/alcotest.ml", line 290, characters 8-14


The full test results are available in `/home/afu/git/ppx_protocol_conv/_build/_tests/835BF394-F1E5-476D-8A51-14C558AC4D4D`.
1 error! in 0.005s. 62 tests run.
```

emillon commented:
From a quick read this looks like what I was after in #146, thanks!

@nathanreb should be interested in reviewing this 😃

andersfugmann commented:
I like the printing of 'E' for errors. I will update the PR to do that.

andersfugmann commented:
I've updated the PR to reflect your comments. I also solved a bug introduced with the change to print 'E' for errors, as I did not notice that the function `error` had the side-effect of storing the stringified error. I've changed it to only have this side effect, and print status lined explicitly. 

I've also renamed the option to be `--compact`

NathanReb commented on src/alcotest.ml:
```diff
@@ -227,38 +226,51 @@ let error t path fmt =
+  | `Exn _   -> print_ch t "F"
+  | `Error _ -> print_ch t "E"
+  | `Skip    -> print_ch t "S"
+  | `Todo _  -> print_ch t "T"
```
Awesome ordering!
